src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -404,10 +407,21 @@ public boolean isChatColorsEnabled() {
     }
 
     @Override
+    public void sendThroughMessage(Text message) {
+        checkNotNull(message, "The message text cannot be null!");
+        this.connection.sendPacket(new SPacketChat(SpongeTexts.toComponent(message)));
+    }
+
+    @Override
     public void sendMessage(ChatType type, Text message) {
         checkNotNull(type, "type");
         checkNotNull(message, "message");
 
+        if (this.conversation != null && this.conversation.getChatHandler(this).isPresent()
+            && !this.conversation.getChatHandler(this).get().process(message)) {
+            return;
+        }
+
         ITextComponent component = SpongeTexts.toComponent(message);
```

Comment:
You should move this logic to a method like `sendMessage0` and call it from `sendMessage` and `sendThroughMessage`.